src/libAtomVM/nifs.c

```diff
@@ -2184,42 +2185,11 @@ static term nif_erlang_atom_to_list_1(Context *ctx, int argc, term argv[])
     return ret;
 }
 
-static size_t lltoa(avm_int64_t int_value, unsigned base, char *integer_string)
+static term integer_to_buf(Context *ctx, int argc, term argv[], char *tmp_buf, size_t tmp_buf_size,
```

Comment:
I am confused about the signature, returned value is [] or invalid_term, isn't it?
Couldn't we return just a bool?

Reply:
ugly signature indeed, that function raises different kinds of errors for both `integer_to_binary` and `integer_to_list`